devito/types/dimension.py

```diff
@@ -950,9 +966,14 @@ def free_symbols(self):
             pass
         return retval
 
-    def _arg_values(self, interval, grid=None, **kwargs):
-        # Parent dimension define the interval
-        fact = self._factor.data if self._factor is not None else 1
+    def _arg_values(self, interval, grid=None, args=None, **kwargs):
+        if self.symbolic_factor is not None:
```

Comment:
assuming it's possible to turn symbolic_factor into a cached_property, here we would need `if self.factor. is not None`